### PR TITLE
[wptserve] Add logging of stacktraces for failed request handlers

### LINKED ISSUE #30070: Errors in wptserve handlers aren't logged effectively
tl;dr: wptserve should print stacktraces when request handlers fail

While working on crbug.com/1218565 I was hitting unexpected test failures that were ultimately caused by a Python error in one of our server-side request handlers (for the curious, https://github.com/web-platform-tests/wpt/blob/21f16b1e2f1b404e79d619c495585c922ee16e0e/resource-timing/resources/TAOResponse.py fails with an UnboundLocalError if the 'origin' request header is missing but the 'tao' query-parameter is set to 'match_origin').

When digging in, I found wptserve's stderr was empty but stdout contained something like
```
[<timestamp>,<pid> http on port 8001] ERROR - Exception loading http://www1.web-platform.test:8001/resource-timing/resources/TAOResponse.py?tao=match_origin&unique=8813844539736879: 500
[<timestamp>,<pid> http on port 8001] INFO - 500
```
I was expecting that, if a handler script threw an unhandled exception, there'd be some kind of stacktrace of what happened in the handler. This would give test authors a clear indication that they made a mistake in their request handler code.

### PR DESCRIPTION
**Pull request by tommckee1**
This PR fixes #30070

### CODE REVIEW

**Comment by tommckee1** [comment]
Although this is a small change, let me know if there needs to be test coverage... adding and defining test coverage for what shows up on stdout/stderr is probably a much bigger/more involved change than this :)

**Comment by tommckee1** [comment]
Looks like the CI failures are timeouts[1]. Not sure if there's a way to retry the failed tests?

I guess it's possible that, if a test was hitting a lot of server-side errors (500 or, more likely, 404) that the additional time to print the stacktraces could add up to some meaningful amount of time.

FWIW, I do see a few 404 stacktraces for favicon.ico[2][3] in the logs which would be new with this patch, but I wouldn't expect a timeout from so few (10 by my count) logging statements. What do others think here?

Another guess would be, if multiple jobs are running on the same instance in parallel, extra time/resources used by extra logging could starve neighbouring jobs? I can imagine some tests have a _lot_ of 404s, so are sensitive to this patch, but I'd expect those tests to timeout directly instead of clobbering a neighbour, right? I would be very surprised if this was the case, but maybe folks with more context would know: Does making TestA more expensive ever lead to TestB timing out?

[1]: https://dev.azure.com/web-platform-tests/wpt/_build/results?buildId=69033&view=logs&j=af75b4de-b26f-5d4d-58e8-b41258076d44&t=052839da-7227-5391-fe2d-4b66ec169a0c&l=768
[2]: https://dev.azure.com/web-platform-tests/wpt/_build/results?buildId=69033&view=logs&j=af75b4de-b26f-5d4d-58e8-b41258076d44&t=052839da-7227-5391-fe2d-4b66ec169a0c&l=85
[3]: https://dev.azure.com/web-platform-tests/wpt/_build/results?buildId=69033&view=logs&j=af75b4de-b26f-5d4d-58e8-b41258076d44&t=052839da-7227-5391-fe2d-4b66ec169a0c&l=384

File: tools/wptserve/wptserve/server.py [hunk 1]
```diff
@@ -285,8 +285,12 @@ def finish_handling(self, request, response, handler):
             try:
                 handler(request, response)
             except HTTPException as e:
+                self.logger.error("HTTPException in handler: %s" % e)
```
Comment (jgraham) [inline tools/wptserve/wptserve/server.py:288]: I think I'd like us to try only logging stacks for 500-type errors (here or below), and at warning level. For other exceptions (e.g. 404) the normal logging should be sufficient.

**Review by jgraham — CHANGES_REQUESTED** [review]

File: tools/wptserve/wptserve/server.py (hunk 1 above)
Comment (tommckee1) [reply tools/wptserve/wptserve/server.py]: I've updated the PR to focus on logging only when the handler throws a 5xx 
code. LMKWYT!

File: tools/wptserve/wptserve/server.py [hunk 2]
```diff
@@ -285,6 +285,9 @@ def finish_handling(self, request, response, handler):
             try:
                 handler(request, response)
             except HTTPException as e:
+                if e.code // 100 == 5:
```
Comment (jgraham) [inline tools/wptserve/wptserve/server.py:288]: This seems unnecessarily clever compared with e.g. `if 500 <= e.code < 600`

**Review by jgraham — CHANGES_REQUESTED** [review]

File: tools/wptserve/wptserve/server.py (hunk 2 above)
Comment (tommckee1) [reply tools/wptserve/wptserve/server.py]: Agreed! Fixed in next commit.

**Review by foolip — APPROVED** [review]

**Comment by foolip** [comment]
This looks sensible and @jgraham's previous comments seem addressed. James, do you want to take another look though?

**Review by jgraham — APPROVED** [review]

**Pull request merged**